storage/beacondb/eth1.go

```diff
@@ -51,8 +55,13 @@ func (kv *KVStore) SetLatestExecutionPayloadHeader(
 }
 
 // GetEth1DepositIndex retrieves the eth1 deposit index from the beacon state.
+// If the index is not found, it returns 0 and no error.
 func (kv *KVStore) GetEth1DepositIndex() (uint64, error) {
-	return kv.eth1DepositIndex.Get(kv.ctx)
+	index, err := kv.eth1DepositIndex.Get(kv.ctx)
+	if errors.Is(err, collections.ErrNotFound) {
+		return 0, nil
+	}
```

Comment:
I think this should be not necessary if we duly set the index for genesis validators?

Reply:
but this is checked in validateGenesisDeposits and right now expects an error. If the error is something other than ErrNotFound we will continue processing even if something else is wrong.